internal/pkg/cli/svc_deploy.go

```diff
@@ -286,38 +290,49 @@ func (o *deploySvcOpts) configureClients() error {
 }
 
 func (o *deploySvcOpts) pushToECRRepo() error {
-	path, err := o.getDockerfilePath()
+
+	dockerBuildInput, err := o.getBuildArgs()
 	if err != nil {
 		return err
 	}
 
-	if err := o.imageBuilderPusher.BuildAndPush(docker.New(), path, o.ImageTag); err != nil {
+	if err := o.imageBuilderPusher.BuildAndPush(docker.New(), dockerBuildInput); err != nil {
 		return fmt.Errorf("build and push image: %w", err)
 	}
 
 	return nil
 }
 
-func (o *deploySvcOpts) getDockerfilePath() (string, error) {
-	type dfPath interface {
-		DockerfilePath() string
+func (o *deploySvcOpts) getBuildArgs() (*docker.BuildArguments, error) {
+	type dfArgs interface {
+		BuildArgs(rootDirectory string) *manifest.DockerBuildArgs
 	}
 
 	manifestBytes, err := o.ws.ReadServiceManifest(o.Name)
 	if err != nil {
-		return "", fmt.Errorf("read manifest file %s: %w", o.Name, err)
+		return nil, fmt.Errorf("read manifest file %s: %w", o.Name, err)
 	}
-
-	svc, err := manifest.UnmarshalService(manifestBytes)
+	svc, err := o.unmarshal(manifestBytes)
 	if err != nil {
-		return "", fmt.Errorf("unmarshal svc manifest: %w", err)
+		return nil, fmt.Errorf("unmarshal service %s manifest: %w", o.Name, err)
 	}
-
-	mf, ok := svc.(dfPath)
+	mf, ok := svc.(dfArgs)
 	if !ok {
-		return "", fmt.Errorf("service %s does not have a dockerfile path", o.Name)
+		return nil, fmt.Errorf("service %s does not have required method Build()", o.Name)
+	}
+	copilotDir, err := o.ws.CopilotDirPath()
+	if err != nil {
+		return nil, fmt.Errorf("get copilot directory: %w", err)
 	}
-	return mf.DockerfilePath(), nil
+	wsRoot := filepath.Dir(copilotDir)
```

Comment:
Is this needed? I believe `CopilotDirPath` returns a directory.

Reply:
Yes, because we want the _workspace root_, not the Copilot dir. `filepath.Dir` is equivalent to the shell `dirname` command.